File: conectaBrenannd-api/conectaBrenannd-api/src/main/java/com/conecta/brenannd/service/IGrupoService.java
package com.conecta.brenannd.service;

import com.conecta.brenannd.entity.Grupo;

public interface IGrupoService {
	
	Grupo save(final Grupo entity);
	
	Grupo update(final long id, final Grupo entity);
	
	void delete(final long id);

}

File: conectaBrenannd-web/src/app/components/entity/grupo/services/igrupo.service.ts
import { Observable } from 'rxjs';
import { Grupo } from '../models/grupo.models';

export interface IGrupoService {
  save(request: Grupo): Observable<Grupo>;

  update(id: number, request: Grupo): Observable<Grupo>;

  delete(id: number): Observable<void>;

  list(): Observable<Grupo[]>;

  findByID(id: number): Observable<Grupo>;

  listByUser(): Observable<Grupo[]>;
}
